addons/website_payment/static/src/snippets/s_supported_payment_methods/supported_payment_methods.js

```diff
@@ -0,0 +1,60 @@
+import { browser } from '@web/core/browser/browser';
+import { rpc } from '@web/core/network/rpc';
+import { registry } from '@web/core/registry';
+import { Interaction } from '@web/public/interaction';
+
+
+export class SupportedPaymentMethods extends Interaction {
+    static selector = '.s_supported_payment_methods';
+
+    setup() {
+        this.payment_methods = [];
+        this.templateKey = 'website_payment.s_supported_payment_methods.icons';
+    }
+
+    async willStart() {
+        await this.fetchPaymentMethods();
+    }
+
+    /**
+     * Fetch the payment methods and cache them in the session.
+     *
+     * Caching limits the amount of rpc calls when switching pages, or when editing the snippet in
+     * the editor as any edit reloads the interaction.
+     */
+    async fetchPaymentMethods() {
+        let cache = JSON.parse(
+            browser.sessionStorage.getItem('website_payment.supported_payment_methods') || '{}',
+        );
+
+        // Re-fetch if the cached list can potentially be larger
+        if (cache.payment_methods === undefined || cache.limit < this.limit) {
+            cache.payment_methods = await this.waitFor(
+                rpc('/website_payment/snippet/supported_payment_methods', { limit: this.limit }),
+            ).catch(_ => []);
+            cache.limit = this.limit;
+            browser.sessionStorage.setItem(
+                'website_payment.supported_payment_methods', JSON.stringify(cache),
+            );
```

Comment:
This prevents displaying any newly enabled/disabled payment method until the tab is closed, making this now dynamic snippet temporarily static 🙈